scipy/stats/_stats_py.py

```diff
@@ -3240,7 +3244,7 @@ def gstd(a, axis=0, ddof=1):
             raise ValueError(w) from w
         else:
             #  Remaining warnings don't need to be exceptions.
-            return np.exp(np.std(log(a, where=~a_nan), axis=axis, ddof=ddof))
+            return xp.exp(xp.std(log(a), axis=axis, **ddof))
```

Comment:
Similarly here. I don't know of circumstances under which taking the log of `nan` emits a warning or returns anything other than `nan`.